x-pack/plugins/security_solution/server/lib/telemetry/fleet_saved_objects.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ISavedObjectsRepository } from 'src/core/server';
+import { AgentEventSOAttributes } from './../../../../ingest_manager/common/types/models/agent';
+import {
+  AGENT_SAVED_OBJECT_TYPE,
+  AGENT_EVENT_SAVED_OBJECT_TYPE,
+} from './../../../../ingest_manager/common/constants/agent';
+import { Agent, DefaultPackages as FleetDefaultPackages } from '../../../../ingest_manager/common';
+
+export const FLEET_ENDPOINT_PACKAGE_CONSTANT = FleetDefaultPackages.endpoint;
+
+export const getFleetSavedObjectsMetadata = async (savedObjectsClient: ISavedObjectsRepository) =>
+  savedObjectsClient.find<Agent>({
+    type: AGENT_SAVED_OBJECT_TYPE,
+    fields: ['packages', 'last_checkin', 'local_metadata'],
+    filter: `${AGENT_SAVED_OBJECT_TYPE}.attributes.packages: ${FLEET_ENDPOINT_PACKAGE_CONSTANT}`,
+  });
+
+export const getFleetEventsSavedObjects = async (
+  savedObjectsClient: ISavedObjectsRepository,
+  agentId: string
+) =>
+  savedObjectsClient.find<AgentEventSOAttributes>({
+    type: AGENT_EVENT_SAVED_OBJECT_TYPE,
+    filter: `${AGENT_EVENT_SAVED_OBJECT_TYPE}.attributes.agent_id: ${agentId}`,
```

Comment:
@jonathan-buttner , is there a way for me to also filter for values that have the value `endpoint` somewhere in the `attributes.message` field? Just trying to think of ways to optimize this here

Reply:
👍 on filtering this, also maybe fetch only data for the last n time, no? this list could be really long at some point

Reply:
Hmm I'm not sure about that. I'll have to take a look at how the filtering works with saved objects.

Reply:
@nchaulet I would do that, but only issue is, if a policy was enabled a week ago, and nothing has changed, then there won't be any `endpoint` related updates in the events lists in the past lets say 24 hours. i.e. there isn't a `STILL_RUNNING` subtype 😂. Hence why it would be great to filter by all references to endpoint, but don't think that's possible via the filter? 

If I were to just check if anything has changed regarding the endpoint in the last 24 hours, then I would have to keep some knowledge from day to day of the previous state, and that feels like a recipe for bugs if something gets missed in the cracks somehow

Reply:
Looks like it's pure EQL and this should work to get all endpoint references in the message! Hoot Hoot 😄 

```
`${AGENT_EVENT_SAVED_OBJECT_TYPE}.attributes.agent_id: ${agentId} and ${AGENT_EVENT_SAVED_OBJECT_TYPE}.attributes.message: "${FLEET_ENDPOINT_PACKAGE_CONSTANT}"`
```